massa-db-worker/src/massa_db.rs

```diff
@@ -180,7 +180,8 @@ where
 
             for (serialized_key, serialized_value) in db_iterator.flatten() {
                 new_elements_size += serialized_value.len();
-                if new_elements_size < self.config.max_final_state_elements_size {
+                new_elements_size += serialized_key.len();
+                if new_elements_size <= self.config.max_final_state_elements_size {
```

Comment:
Can you do the same change (Add the key length to the total size value) in the `get_versioning_batch_to_stream` just below?

Reply:
Done

Reply:
Feel free to re-resolve this if I'm mistaken, but I think the compare in versioning should be <= to be consistant with the coppare in state?

Reply:
DOne